### ISSUE TITLE
[NEW ROADMAP] Machine Learning

### ISSUE THREAD

**Issue by letsmakecakes** [kamranahmedse/developer-roadmap#4003]
Machine Learning Roadmap for roadmap.sh:

1. Productivity and Learning Techniques:
   - Read "Atomic Habits: An Easy & Proven Way to Build Good Habits & Break Bad Ones" by James Clear.
   - Read "Deep Work" by Cal Newport.
   - Take the "Learning How to Learn: Powerful mental tools to help you master tough subjects" course on Coursera.

2. Programming Languages:
   - Start with Python:
     - Complete the "Python for Everybody Specialization" on Coursera.
     - Follow Krish Naik's Python Tutorial on YouTube.
     - Read "Learn Python the Hard Way" by Zed Shaw.
   - Explore other programming languages:
     - R Programming on Coursera.
     - SAS Programmer Professional Certificate on Coursera.
     - Functional Programming in Scala Specialization on Coursera.
     - Julia Scientific Programming on Coursera.
     - Introductory C Programming Specialization on Coursera.
     - Coding for Everyone: C and C++ Specialization on Coursera.

3. Mathematics:
   - Statistics:
     - Watch Krish Naik's Statistics Playlist on YouTube.
     - Learn statistics and probability on Khan Academy.
     - Complete the FreeCodeCamp Statistics Course on YouTube.
     - Read "Naked Statistics" by Charles Wheelan.
   - Linear Algebra:
     - Learn linear algebra on Khan Academy.
   - Matrices:
     - Study matrices on Khan Academy.
   - Calculus:
     - Learn single-variable calculus on Khan Academy.
     - Study vectors on Khan Academy.
     - Explore multivariable calculus on Khan Academy.
     - Learn differential calculus on Khan Academy.
   - Discrete Mathematics:
     - Complete the Discrete Mathematics specialization on Coursera.
   - Geometry:
     - Study geometry on Khan Academy.
   - Optional: Read the "Mathematics for Machine Learning" book.

4. Web Development Skills:
   - Responsive Web Design on FreeCodeCamp.
   - JavaScript Algorithms and Data Structures on FreeCodeCamp.
   - Front End Development Libraries on FreeCodeCamp.
   - APIs and Microservices on FreeCodeCamp.

5. Database Knowledge:
   - SQL:
     - Watch Freecodecamp SQL on YouTube.
     - Complete the "Intro to SQL" and "Advanced SQL" courses on Kaggle.
     - Learn SQL Basics for Data Science specialization on Coursera.
     - Take LinkedIn Learning courses on becoming a Database Developer and mastering SQL.
   - Oracle SQL:
     - Enroll in the "Oracle SQL: An Introduction to the most popular database" course on Udemy.
   - Apache Cassandra:
     - Learn Apache Cassandra from Scratch on Eduonix or Javatpoint.
   - MongoDB:
     - Explore MongoDB courses on the MongoDB University website.

6. Data Science Foundations:
   - Complete the "Intro to Data Science" course on Udacity.

7. Pandas:
   - Learn Pandas on Kaggle.

8. Data Cleaning:
   - Learn Pythonic Data Cleaning with Pandas and NumPy.
   - Take Edureka's Microsoft Excel course.
   - Complete the Data Cleaning course on Kaggle.

9. Data Analysis:
   - Inferential Statistics on Coursera.
   - IBM Data Analyst Professional Certificate on Coursera.
   - Take LinkedIn Learning courses on becoming a Data Analyst.
   - Machine Learning Pipelines with Azure ML Studio project on Coursera.
   - Data Analysis with Python on FreeCodeCamp.

10. Exploratory Data Analysis:
    - Watch "Intro To EDA" by Code Heroku on YouTube.
    - Read the "Exploratory Data Analysis: Iris Dataset" article.
    - Complete the "Probability Theory, Statistics, and Exploratory Data Analysis"

11. Data Visualization:
   - Learn Data Visualization with Python on FreeCodeCamp.
   - Complete the "Data Visualization with Python" course on Coursera.
   - Master Tableau Desktop on Tableau's official website.
   - Explore the "Data Visualization and Communication with Tableau" course on Coursera.

12. Machine Learning:
   - Machine Learning by Andrew Ng on Coursera.
   - Kaggle Machine Learning micro-courses.
   - "Hands-On Machine Learning with Scikit-Learn, Keras, and TensorFlow" book by Aurélien Géron.
   - "Python Machine Learning" book by Sebastian Raschka and Vahid Mirjalili.

13. Deep Learning:
   - Deep Learning Specialization by Andrew Ng on Coursera.
   - TensorFlow Developer Professional Certificate on Coursera.
   - "Deep Learning" book by Ian Goodfellow, Yoshua Bengio, and Aaron Courville.

14. Natural Language Processing (NLP):
   - Natural Language Processing Specialization on Coursera.
   - "Speech and Language Processing" book by Daniel Jurafsky and James H. Martin.
   - "Natural Language Processing with Python" book by Steven Bird, Ewan Klein, and Edward Loper.

15. Reinforcement Learning:
   - Reinforcement Learning Specialization on Coursera.
   - "Reinforcement Learning: An Introduction" book by Richard S. Sutton and Andrew G. Barto.

16. Model Deployment:
   - Deploy Machine Learning Models using Flask on FreeCodeCamp.
   - Deploying Machine Learning Models with AWS on Coursera.
   - Deploying Machine Learning Models using TensorFlow Serving on Coursera.

17. Big Data Technologies:
   - Apache Hadoop:
     - Learn Apache Hadoop on Hadoop's official website.
   - Apache Spark:
     - Apache Spark with Python - Big Data with PySpark and Spark course on Udemy.
     - "Learning Spark" book by Jules S. Damji, Brooke Wenig, Tathagata Das, and Denny Lee.

18. Cloud Computing Platforms:
   - Amazon Web Services (AWS):
     - AWS Certified Machine Learning - Specialty certification.
     - "AWS Certified Machine Learning - Specialty (MLS-C01) Exam Guide" book by Vinoth Kumar.
   - Google Cloud Platform (GCP):
     - Google Cloud Certified - Professional Data Engineer certification.
     - "Google Cloud Certified - Professional Data Engineer Study Guide" book by Dan Sullivan.
   - Microsoft Azure:
     - Microsoft Certified: Azure AI Engineer Associate certification.
     - "Exam AI-100: Designing and Implementing an Azure AI Solution" book by Scott Duffy.

19. Continuous Learning and Projects:
   - Participate in Kaggle competitions.
   - Join open-source projects on GitHub.
   - Create your own machine learning projects.

This roadmap serves as a general guide, and anyone can adapt it based on their specific interests and goals.

**Comment by XmySz** [comment]
good job!

**Comment by kamranahmedse** [comment]
Hey there!

Looks like this issue has been hanging around for a bit without much action. Our roadmaps have evolved quite a bit since then, and a bunch of older issues aren't really applicable anymore. So, we're tidying things up by closing out the older ones to keep our issue tracker nice and organized for future feedback.

If you still think this problem needs addressing, don't hesitate to reopen the issue. We're here to help!

Thanks a bunch!

**Comment by sohang3112** [comment]
@letsmakecakes Hi, this Machine Learning roadmap you made looks quite good! Did you upload it somewhere in the roadmap.sh site also?  I couldn't find it in both official roadmaps and community roadmaps.

**Comment by letsmakecakes** [comment]
> @letsmakecakes Hi, this Machine Learning roadmap you made looks quite good! Did you upload it somewhere in the roadmap.sh site also?  I couldn't find it in both official roadmaps and community roadmaps.

No, I did not upload it anywhere else. It still needs some tweaks here and there.
